presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java

```diff
@@ -282,13 +292,12 @@ public Node getNodeForBucket(int bucketId)
         }
     }
 
-    private static class LifespanScheduler
+    private static class FixedLifespanScheduler
+            implements LifespanScheduler
     {
-        // Thread Safety:
-        // * Invocation of onLifespanFinished can be parallel and in any thread.
-        //   There may be multiple invocations in flight at the same time,
-        //   and may overlap with any other methods.
-        // * Invocation of all other methods happens sequentially in a single thread.
+        /**
+         * See {@link LifespanScheduler} about thread safety
```

Comment:
If you want JavaDoc comment for this, move this above `FixedLifespanScheduler`, otherwise it will be attached to the first field in this class (`driverGroupToNodeMap`)